### PR TITLE
Colorization

### PR DESCRIPTION
**Pull request by SphericalPotatoInVacuum**

### CODE REVIEW

File: src/config/color.py [hunk 1]
```diff
@@ -1,56 +1,67 @@
...
+        model = RRDBNet(3, 3, 64, 10, gc=32).to(device)
         criterion = torch.nn.MSELoss(reduction='mean').to(device)
-        optimizer = torch.optim.Adam(model.parameters(), lr=1e-3, weight_decay=1e-6)
+        optimizer = torch.optim.Adam(model.parameters(), lr=1e-4, weight_decay=1e-6)
         scheduler = torch.optim.lr_scheduler.MultiStepLR(
             optimizer,
             milestones=list(range(0, total_epochs, 5)),
-            gamma=0.2
+            gamma=0.2,
+            verbose=True
```
Comment (nikoryagin) [inline src/config/color.py:35]: Maybe it would be better to add info about scheduler step to our Logger? otherwise it could be lost in tons of other messages

File: src/config/color.py [hunk 2]
```diff
@@ -1,56 +1,67 @@
...
         )
 
-        batch_size = 6
-
-        train_dataset = CustomDataset(Path('./data/raw/AlsoCoco/train2017'))
+        batch_size = 56
```
Comment (nikoryagin) [inline src/config/color.py:38]: I recommend you to research a little bit about batch size being power of 2. It's highly likely that with batch_size=32 would work faster than batch_size=56 due to some issues with virtual processors afaik. I tried to google it, but i think you will understand it better. here's link to start https://datascience.stackexchange.com/questions/20179/what-is-the-advantage-of-keeping-batch-size-a-power-of-2

Comment (SphericalPotatoInVacuum) [reply src/config/color.py:38]: Color me surprised: I did not expect that setting `batch_size=32` would make training an epoch 28% faster 😲

File: src/training/color.py [hunk 3]
```diff
@@ -32,113 +28,78 @@ def __init__(self,
...
+    def train(self):
+        self.model.train()
 
-            # x: l_img x3
-            # y: ab_img
-            for idx, (l_img, ab_img) in enumerate(self.train_dataloader):
-                # Skip bad data
-                if not l_img.ndim:
-                    continue
+        # inputs: l_img x3
```
Comment (grenlayk) [inline src/training/color.py:34]: `l_img` -> `bw_img` as we use this naming in dataset

File: src/logger/simple.py [hunk 4]
```diff
@@ -22,7 +22,7 @@ def __init__(self, print_freq: int = 100, image_freq: int = 1000, wb_path: str =
         self.wb_path: str = wb_path or "./wandb"
         self.train_iter = 1
         self.val_iter = 1
-        self.wandb = wandb.init(project=project_name, entity="text-deep-fake")
+        self.wandb = wandb.init(project=project_name, entity="sphericalpotatoinvacuum")
```
Comment (grenlayk) [inline src/logger/simple.py:25]: Do not commit your personal entity pls

File: src/config/color.py [hunk 5]
```diff
@@ -1,56 +1,75 @@
...
+        if not data_path.exists():
+            disk.download(data_path.with_suffix('.zip'), data_path.with_suffix('.zip'))
+            with zipfile.ZipFile(data_path.with_suffix('.zip'), 'r') as zip_ref:
+                zip_ref.extractall('data/')
 
-        train_dataset = CustomDataset(Path('./data/raw/AlsoCoco/train2017'))
+        total_epochs = 20  # 20
+
+        m = resnet18(True)
+        m = nn.Sequential(*list(m.children())[:-2])
```
Comment (grenlayk) [inline src/config/color.py:34]: Why do you use `m` variable twice? and what does it stands for?
maybe 
```python
name_that_tells_something = nn.Sequential(*list(resnet18(True).children())[:-2])
model = DynamicUnet(name_that_tells_something, 3, (128, 128)).to(device)`
```

File: src/config/color.py [hunk 6]
```diff
@@ -1,56 +1,75 @@
+import zipfile
 from pathlib import Path
 
 import torch
+from fastai.vision.models import resnet18
+from fastai.vision.models.unet import DynamicUnet
+from loguru import logger
 from src.data.color import CustomDataset
```
Comment (grenlayk) [inline src/config/color.py:8]: I suddenly understood that we have kind a confusion in naming. I suppose this custom dataset should be called `ColorDataset` or smth like this. (Same as we have `BaselineDataset` in baseline branch and `SimpleDataset` for template objects.)

What do you think about it? If you agree I will simply change it in my branch (colorization-plus-gan).

File: src/config/color.py [hunk 7]
```diff
@@ -1,56 +1,75 @@
...
-            test_dataset,
             batch_size=1,
-            shuffle=False,
-            num_workers=8
+            shuffle=False
         )
+        logger.info(f'Validate size: {len(val_dataloader)} x {1}')
 
-        point_batches = 500
+        test_dataloader = None
```
Comment (grenlayk) [inline src/config/color.py:59]: I think you can simply delete it and do not pass anything to the `Trainer`.

File: src/config/color.py (hunk 6 above)
Comment (SphericalPotatoInVacuum) [reply src/config/color.py:8]: Yes it makes sense to have different names if they do different things

File: src/logger/simple.py [hunk 8]
```diff
@@ -49,9 +49,10 @@ def log_train(self, losses: Optional[Dict[str, float]] = None, images: Optional[
             self.loss_buff['sumlast'].clear()
 
         if self.train_iter % self.image_freq == 0 and images:
-            for image_name, image in images.items():
-                self.wandb.log({f'{image_name}': wandb.Image(image)})
+            self.wandb.log({f'train/{image_name}': wandb.Image(image)
+                           for image_name, image in images.items()}, commit=False)
 
+        self.wandb.log({}, commit=True)
```
[the remark this answers is not in the document]
Comment (SphericalPotatoInVacuum) [reply src/logger/simple.py:55]: It is to send all of the logs that were logged before as one step. All of the previous `log` calls have `commit=False` to not send that log as a separate step.

File: src/losses/compose.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
+from typing import List
+
+import torch
+
+
+class Compose(torch.nn.Module):
```
[the remark this answers is not in the document]
Comment (SphericalPotatoInVacuum) [reply src/losses/compose.py:6]: Yeah that would make sense. I decided to steal the idea from torchvision transforms :)
